### PR TITLE
Keep generated JS/JSX code examples in the repo

### PR DESCRIPTION
**Pull request by sequba**
### Context

- [x] added npm script that generates a single JS/JSX code example
- [x] modify README to describe the flow
- [x] add generated JS/JSX to the repository
- [x] modify `.gitignore` accordingly
- [x] add GH action that verifies if the generated JS/JSX versions of the examples match the committed ones

Dev workflow when modifying any code example:
```
1. modify content/guides/some/example.ts file
2. run `npm run docs:scripts:generate-js-example content/guides/some/example.ts (generates content/guides/some/example.js)
3. commit both content/guides/some/example.ts and content/guides/some/example.js
4. make sure that GH action "Check docs examples" reports no issues
```

### How has this been tested?

- [x] verify npm scripts locally
- [x] verify that GH action works correctly when run on PR:
  - [x] with no changes to examples
  - [x] with TS example changed but JS version not updated
  - [x] with TS example changed and JS version updated correctly

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
https://github.com/handsontable/dev-handsontable/issues/1944

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [ ] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

[skip changelog]

### CODE REVIEW

**Comment by evanSe** [comment]
@sequba rather than write code that does a diff check couldn't we just do this inside of an action?
```
steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - name: Check for Git changes
        id: git_changes
        run: |
          if git diff --exit-code && git diff --cached --exit-code; then
            echo "changed=false" >> $GITHUB_ENV
          else
            echo "changed=true" >> $GITHUB_ENV
          fi
```

File: docs/README.md [hunk 1]
```diff
@@ -25,15 +25,22 @@ To start a local Handsontable documentation server:
...
-5. In your browser, go to: http://localhost:8080/docs/.
+4. In your browser, go to: http://localhost:8080/docs/.
+
+## Handsontable documentation code examples
+
+Most code examples in the `content/guides` directory are kept in 2 versions: TS/TSX and JS/JSX. When modifying the code example or adding a new one, you should always update the TS/TSX version and then generate the JS/JSX version:
+
+E.g.:
+1. Modify `content/guides/some/example.ts` file.
+2. Run `npm run docs:scripts:generate-js-example content/guides/some/example.ts` to generate `content/guides/some/example.js`.
```
Comment (sequba) [inline docs/README.md:40]: TODO: update this line

File: docs/README.md [hunk 2]
```diff
@@ -52,7 +59,8 @@ From the `docs` directory, you can run the following npm scripts:
 * `npm run docs:scripts:link-assets` – Prepares the `next` documentation version's CSS and JavaScript.
 * `npm run docs:review [COMMIT_HASH]` – Deploys the documentation locally at a `[COMMIT_HASH]` commit.
 * `npm run docs:test:example-checker` – Runs the tests that checks if all Docs examples work.
-* `npm run docs:scripts:generate-js-examples` – Generate js version of the code examples (need to run this script before first `docs:start` and every time some `*.ts` inside `docs/content/guides` changes)
+* `npm run docs:scripts:generate-js-example content/guides/path/to/example.ts` – Generate JS/JSX version of the code example (needs to be run before commiting any change to TS/TSX code example)
+* `npm run docs:scripts:generate-js-example:all` – Generate all JS/JSX versions of the TS/TSX code examples in content/guides/ directory
```
Comment (sequba) [inline docs/README.md:63]: TODO: update scripts

File: docs/package.json [hunk 3]
```diff
@@ -14,12 +14,15 @@
...
+    "docs:code-examples:generate-all-js": "bash .vuepress/tools/code-examples-generator.sh --generateAll",
+    "docs:code-examples:verify-all-js": "bash .vuepress/tools/code-examples-generator.sh --verifyAll",
+    "docs:code-examples:format-all-ts": "bash .vuepress/tools/code-examples-generator.sh --formatAllTsExamples",
     "docs:start": "npm run docs:scripts:link-assets && cross-env NODE_OPTIONS=--openssl-legacy-provider node ./node_modules/.bin/vuepress dev ./",
     "docs:start:no-cache": "npm run docs:scripts:link-assets && cross-env NODE_OPTIONS=--openssl-legacy-provider node ./node_modules/.bin/vuepress dev --no-cache ./",
     "docs:serve": "http-server -c-1 .vuepress/dist/",
-    "docs:build": "npm run docs:scripts:generate-js-examples && npm run docs:build:no-examples",
-    "docs:build:no-examples": "cross-env NODE_OPTIONS=--openssl-legacy-provider node .vuepress/tools/build.mjs -- --no-cache",
+    "docs:build": "cross-env NODE_OPTIONS=--openssl-legacy-provider node .vuepress/tools/build.mjs -- --no-cache",
+    "docs:tmp": "cross-env NODE_OPTIONS=--openssl-legacy-provider node ./node_modules/vuepress/cli.js dev --no-cache ./",
```
Comment (sequba) [inline docs/package.json:25]: TODO: remove

File: .github/workflows/docs-examples.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+env:
+  NODE_VERSION: 20
+
+jobs:
+  docs_examples:
+    name: Docs Examples
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout pull request HEAD branch
```
Comment (sequba) [inline .github/workflows/docs-examples.yml:15]: TODO: check out the merge commit (the default behavior)

File: .github/workflows/docs-examples.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        run: |
+          npm ci --no-audit
+      
+      - name: Generate JS code examples
+        run: |
+          cd docs/ && npm run docs:code-examples:generate-all-js
+
+      - name: Check if generated examples made any changes
+        run: |
+          if git diff --exit-code; then
```
Comment (sequba) [inline .github/workflows/docs-examples.yml:40]: TODO: `git diff --exit-code && git diff --cached --exit-code`

**Comment by sequba** [comment]
> couldn't we just do this inside of an action?
> 
> ```yaml
>       - name: Check for Git changes
>         run: |
>           if git diff --exit-code && git diff --cached --exit-code; then
>             echo "No changes detected"
>           else
>             exit 1
>           fi

@evanSe Good idea. I tested it - it works fine and, at the same time, is more straightforward than my approach. I modified the GH workflow yaml.

File: docs/README.md (hunk 1 above)
Comment (sequba) [reply docs/README.md]: done

File: docs/README.md (hunk 2 above)
Comment (sequba) [reply docs/README.md]: done

File: docs/package.json (hunk 3 above)
Comment (sequba) [reply docs/package.json]: done

File: .github/workflows/docs-examples.yml (hunk 4 above)
Comment (sequba) [reply .github/workflows/docs-examples.yml]: done

File: .github/workflows/docs-examples.yml (hunk 5 above)
Comment (sequba) [reply .github/workflows/docs-examples.yml]: done

File: docs/package.json [hunk 6]
```diff
@@ -14,12 +14,13 @@
...
     "docs:scripts:generate-legacy-docs-versions": "node .vuepress/tools/generate-legacy-docs-versions.mjs",
-    "docs:scripts:generate-js-examples": "bash .vuepress/tools/generate-js-examples.sh",
+    "docs:code-examples:generate-js": "bash .vuepress/tools/code-examples-generator.sh",
+    "docs:code-examples:generate-all-js": "bash .vuepress/tools/code-examples-generator.sh --generateAll",
+    "docs:code-examples:format-all-ts": "bash .vuepress/tools/code-examples-generator.sh --formatAllTsExamples",
     "docs:start": "npm run docs:scripts:link-assets && cross-env NODE_OPTIONS=--openssl-legacy-provider node ./node_modules/.bin/vuepress dev ./",
     "docs:start:no-cache": "npm run docs:scripts:link-assets && cross-env NODE_OPTIONS=--openssl-legacy-provider node ./node_modules/.bin/vuepress dev --no-cache ./",
     "docs:serve": "http-server -c-1 .vuepress/dist/",
-    "docs:build": "npm run docs:scripts:generate-js-examples && npm run docs:build:no-examples",
-    "docs:build:no-examples": "cross-env NODE_OPTIONS=--openssl-legacy-provider node .vuepress/tools/build.mjs -- --no-cache",
```
Comment (evanSe) [inline docs/package.json:22]: Are we sure we want to remove this one?

**Review by evanSe — APPROVED** [review]
In general looks good, the build is failing as expected so we can regenerate the files and merge.

File: docs/package.json (hunk 6 above)
Comment (sequba) [reply docs/package.json:22]: We don't need it anymore since I remove the build examples step from the normal build script

**Pull request merged**